pkg/capabilities/capabilities.go

```diff
@@ -0,0 +1,263 @@
+	UnregisterTrigger(ctx context.Context, request CapabilityRequest) error
+}
+
+// TriggerCapability interface needs to be implemented by all trigger capabilities.
+type TriggerCapability interface {
+	BaseCapability
+	TriggerExecutable
+}
+
+// ActionCapability interface needs to be implemented by all action capabilities.
+type ActionCapability interface {
+	BaseCapability
+	CallbackExecutable
+}
+
+// ConsensusCapability interface needs to be implemented by all consensus capabilities.
+type ConsensusCapability interface {
+	BaseCapability
+	CallbackExecutable
+}
+
+// TargetsCapability interface needs to be implemented by all target capabilities.
+type TargetCapability interface {
+	BaseCapability
+	CallbackExecutable
+}
+
+// CapabilityInfo is a struct for the info of a capability.
+type CapabilityInfo struct {
+	ID             string
+	CapabilityType CapabilityType
+	Description    string
+	Version        string
+}
+
+// Info returns the info of the capability.
+func (c CapabilityInfo) Info(ctx context.Context) (CapabilityInfo, error) {
+	return c, nil
+}
+
+var idRegex = regexp.MustCompile(`[a-z0-9_\-:]`)
+
+const (
+	// TODO: this length was largely picked arbitrarily.
+	// Consider what a realistic/desirable value should be.
+	// See: https://smartcontract-it.atlassian.net/jira/software/c/projects/KS/boards/182
+	idMaxLength = 128
+)
+
+// NewCapabilityInfo returns a new CapabilityInfo.
+func NewCapabilityInfo(
+	id string,
+	capabilityType CapabilityType,
+	description string,
+	version string,
+) (CapabilityInfo, error) {
+	if len(id) > idMaxLength {
+		return CapabilityInfo{}, fmt.Errorf("invalid id: %s exceeds max length %d", id, idMaxLength)
+	}
+	if !idRegex.MatchString(id) {
+		return CapabilityInfo{}, fmt.Errorf("invalid id: %s. Allowed: %s", id, idRegex)
+	}
+
+	if ok := semver.IsValid(version); !ok {
+		return CapabilityInfo{}, fmt.Errorf("invalid version: %+v", version)
+	}
+
+	if err := capabilityType.IsValid(); err != nil {
+		return CapabilityInfo{}, err
+	}
+
+	return CapabilityInfo{
+		ID:             id,
+		CapabilityType: capabilityType,
+		Description:    description,
+		Version:        version,
+	}, nil
+}
+
+// MustNewCapabilityInfo returns a new CapabilityInfo,
+// panicking if we could not instantiate a CapabilityInfo.
+func MustNewCapabilityInfo(
+	id string,
+	capabilityType CapabilityType,
+	description string,
+	version string,
+) CapabilityInfo {
+	c, err := NewCapabilityInfo(id, capabilityType, description, version)
+	if err != nil {
+		panic(err)
+	}
+
+	return c
+}
+
+// TODO: this timeout was largely picked arbitrarily.
+// Consider what a realistic/desirable value should be.
+// See: https://smartcontract-it.atlassian.net/jira/software/c/projects/KS/boards/182
+var defaultExecuteTimeout = 10 * time.Second
+
+// ExecuteSync executes a capability synchronously.
+// We are not handling a case where a capability panics and crashes.
```

Comment:
what does this comment imply? if a panic occurs, what part of the system is taken offline?

Reply:
This comment just intends to clarify that we won't handle exceptional error conditions (eg. a capability crashing); i.e. we can assume that won't happen.